AK/Time.h

```diff
@@ -151,6 +155,24 @@ class Time {
     {
     }
 
+    template<typename T>
+    ALWAYS_INLINE static constexpr i32 sane_mod(T& numerator, i32 denominator)
```

Comment:
Making this a template means making overflow-bugs harder to see, because you're effectively reading two functions at the same time.

Case in point: `from_milliseconds(i64 milliseconds)` instantiates this with `i64`, and might call this with `(0x7fff'ffff'ffff'ffff, 1000)`. Then it ends up trying to write `0x20'c49b'a5e3'53f7` into `i32 dividend`, which overflows.

I *think* this can be fixed by making `dividend` and the return type `T`, especially since the callsites already assume a i64 return value.

Is there a way to say "Instantiate the template like this but only for these specific, hand-selected types"? Something like `static_assert(any_of<i32, i64>(T), "Fragile code ahead! Read it first, and make sure it is overflow-free for your new type before adding to the list.");`